src/main/java/Duke.java

```diff
@@ -1,10 +1,154 @@
+                }
+                case "done": {
+
+                    int taskNum = Integer.parseInt(words[i + 1]);
+                    list[taskNum - 1].maskAsDone();
+                    System.out.println("Nice! I've marked this task as done:"
+                            + "\n"
+                            + "["
+                            + list[taskNum - 1].getStatusIcon()
+                            + "] "
+                            + list[taskNum - 1].getFullDescription()
+                            + BORDER
+                    );
+                    commandsDetected = true;
+                    break;
+                }
+                case "todo": {
+                    String description = "";
+                    for (int j = 1; j < words.length; j++) {
+                        description += words[j] + " ";
+                    }
+                    ToDo t = new ToDo(description);
+                    addToList(t);
+                    commandsDetected = true;
+                    break;
+                }
+                case "deadline":{
+                    int byPosition=0;
+                    String byDescription="";
+                    String description = "";
+                    for(int j =1; j< words.length;j++){
+                        if(words[j].contains("/by")){
+                            byPosition=j;
+                           break;
+                        }
+                        else{
+                            description+= words[j]+ " ";
+                        }
+                    }
+                    for (int k = byPosition+1; k < words.length; k++) {
+                        byDescription= byDescription +" "+ words[k];
+                    }
+                    Deadline d= new Deadline(description,byDescription);
+                    addToList(d);
+                    commandsDetected = true;
+                    break;
+                }
+                case "event":{
+                    int atPosition=0;
+                    String atDescription="";
+                    String description = "";
+                    for(int j =1; j< words.length;j++){
+                        if(words[j].contains("/at")){
+                            atPosition=j;
+                            break;
+                        }
+                        else{
+                            description+= words[j]+ " ";
+                        }
+                    }
+                    for (int k = atPosition+1; k < words.length; k++) {
+                        atDescription= atDescription +" "+ words[k];
+                    }
+                    Event e= new Event(description,atDescription);
+                    addToList(e);
+                    commandsDetected = true;
+                    break;
+                }
+
+                default:{
+                    System.out.println("No proper Commands Detected");
+                    break;
+                }
+                }
```

Comment:
perhaps, you could break up this code into a few other methods? e.g. one method to getInput, another one to printMessage, etc. this will then improve your code's readability (i see lots of deep nesting at the moment) :)